pmd-core/src/main/java/net/sourceforge/pmd/PMDConfiguration.java

```diff
@@ -577,14 +584,23 @@ public AnalysisCache getAnalysisCache() {
     
     /**
      * Sets the analysis cache to be used. Setting a
-     * value of <code>null</code> will cause a Noop AnalysisCache to be used.
+     * value of {@code null} will cause a Noop AnalysisCache to be used.
+     * If incremental analysis was explicitly disabled ({@link #isIgnoreIncrementalAnalysis()}),
+     * then this method is a noop.
      * 
      * @param cache The analysis cache to be used.
      */
     public void setAnalysisCache(final AnalysisCache cache) {
-        if (cache == null) {
+        if (cache == null && isAnalysisCacheFunctional()) {
             analysisCache = new NoopAnalysisCache();
-        } else {
+
+            // Log warning only once, if not explicitly disabled
+            if (!isIgnoreIncrementalAnalysis() && LOG.isLoggable(Level.WARNING)) {
+                final String version = PMDVersion.isUnknown() || PMDVersion.isSnapshot() ? "latest" : "pmd-" + PMDVersion.VERSION;
+                LOG.warning("This analysis could be faster, please consider using Incremental Analysis: "
```

Comment:
I find it very unintuitive to find this here... I'd suggest moving the log to `PMD.doPMD`, which is guaranteed to run once and a little more natural

Reply:
Good idea